04.class/operateInterface.js

```diff
@@ -0,0 +1,88 @@
+import readline from "readline";
+import enquirer from "enquirer";
+
+const { Select } = enquirer;
+
+export default class OperateInterface {
+  constructor(memoOperation) {
+    this.memoOperation = memoOperation;
+  }
+
+  async listMemos() {
+    const memos = await this.memoOperation.fetchMemos();
+    if (this.#isMemoEmpty(memos)) return;
+    memos.forEach((memo) => {
+      console.log(memo.content.split("\n")[0]);
+    });
+  }
+
+  async readMemos() {
+    const content = await this.#selectMemoFromList("see");
+    if (!content) return;
+    console.log(content);
+  }
+
+  async deleteMemo() {
+    const id = await this.#selectMemoFromList("delete");
+    if (!id) return;
+    this.memoOperation.removeMemo(id);
+    console.log("Memo deleted!");
+  }
+
+  async writeMemo() {
+    const content = await this.#imputMemo();
+    await this.memoOperation.addMemo(content);
+    console.log("Memo added.");
+  }
+
+  async #selectMemoFromList(action) {
+    const memos = await this.memoOperation.fetchMemos();
+    if (this.#isMemoEmpty(memos)) return;
+    const choices = this.#extractFirstLineAndValue(memos, action);
```

Comment:
https://github.com/hirokiej/js-practices/pull/5#discussion_r1903172595 にも書いてたのですが、左辺と右辺の名前はなるべくあわせた方が理解しやすいです（memosの中にある選択肢を取り出すわけではないので、extractではなくbuildとしています）

```suggestion
    const choices = this.#buildChoices(memos, action);
```

https://bootcamp.fjord.jp/pages/readable-variable-name

Reply:
失礼しました💦確認漏れでした。
修正します！